src/test/java/software/amazon/awssdk/crt/test/SigningTest.java

```diff
@@ -271,6 +274,53 @@ public Credentials getCredentials() {
         }
         assertTrue(failed);
     }
+    @Test
+    public void testSigningAnonymous() throws Exception {
+            HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+            try (AwsSigningConfig config = new AwsSigningConfig()) {
+                config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+                config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+                config.setRegion("us-east-1");
+                config.setService("service");
+                config.setTime(System.currentTimeMillis());
+                config.setUseDoubleUriEncode(true);
+                config.setShouldNormalizeUriPath(true);
+                config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+                config.setCredentials(Credentials.createAnonymousCredentials());
+
+                CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+                HttpRequest signedRequest = result.get();
+                assertNotNull(signedRequest);
+                assertFalse(hasHeader(signedRequest, "Authorization"));
+            }
+    }
+
+    @Test(expected = CrtRuntimeException.class)
+    public void testSigningNullFail() throws Exception {
+        HttpRequest request = createSimpleRequest("https://www.example.com", "POST", "/derp", "<body>Hello</body>");
+        try{
+            AwsSigningConfig config = new AwsSigningConfig();
+            config.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4);
+            config.setSignatureType(AwsSigningConfig.AwsSignatureType.HTTP_REQUEST_VIA_HEADERS);
+            config.setRegion("us-east-1");
+            config.setService("service");
+            config.setTime(System.currentTimeMillis());
+            config.setUseDoubleUriEncode(true);
+            config.setShouldNormalizeUriPath(true);
+            config.setSignedBodyValue(AwsSigningConfig.AwsSignedBodyValue.EMPTY_SHA256);
+
+            CompletableFuture<HttpRequest> result = AwsSigner.signRequest(request, config);
+            HttpRequest signedRequest = result.get();
+            assertNotNull(signedRequest);
+            assertFalse(hasHeader(signedRequest, "Authorization"));
```

Comment:
Are you trying to assert that an exception IS thrown?
Because this test will pass when no exception is raised

Reply:
It won't pass because of the expected exception in test annotation. I have added some comments to make it more clear. Please take a look and let me know if it is still confusing. Thanks.